packages/backend/src/queryBuilder.ts

```diff
@@ -118,6 +141,12 @@ export const buildQuery = ({
     const dimensionSelects = dimensions.map((field) => {
         const alias = field;
         const dimension = getDimensionFromId(field, explore);
+
+        assertValidDimensionRequiredAttribute(
```

Comment:
Do we need to check filters as well? If I know the dimension, I can get valuable information using filters.  
If we did `filterDimensionsFromExplore(explore, userAttributes)` at the top, we would cover all direct references ( filters, select, order). It still won't cover custom sql in metrics and table calculations.